util/ibex_config.py

```diff
@@ -152,28 +215,40 @@ def get_config_file_location():
 
 
 def main():
-    config_outputs = {'fusesoc_opts': _config_dict_to_fusesoc_opts}
+    outputters = [
```

Comment:
This seems reasonable enough, but have you considered doing it "the other way", so the command line would specify a tool (vcs, riviera or fusesoc) and an action (build, simulate)? That feels like it might be cleaner for command line handling, and would probably be nicer to extend with new tools in the future.

If I've missed something and this is a horrible idea, no worries!

Reply:
I shied away from that mostly because for fusesoc and vcs there are no multiple actions, it's only riviera that needs the two different option sets. I felt it would just complicate matters without really adding anything.

Reply:
I'm not convinced this is simpler, but if you prefer it, no problem.